targets/lenovo-x1-carbon.nix

```diff
@@ -149,29 +150,45 @@
           "pa,id=pa1,server=unix:/run/pulse/native"
         ];
       }
-      ({pkgs, ...}: {
+      ({pkgs, ...}: let
```

Comment:
With the TCP port open anyone can inject "any" data to the TCP port.
With any data injection of random commands becomes possible.
Basename does not do much sanitation for the file, it only removes the path from file and this might make it posible to inject arbitrary commands at least to the last two "openssh ... <path>" statements
I did not test this is practice but if the send to TCP port 1200 is `"filename\"; echo hello>file2"` there might be "file2 with content "hello" in zathura vm?

Reply:
Well, all parameters are wrapped with quotes in the script so I guess when it gets a malformed file path it should fail to copy the file and stop the execution but I think it's a good idea to test different scenarios here. It's actually quite easy to do: ssh to chromium-vm and execute "openpdf some_file".

Reply:
Isn't this openPdf script below at guivm and executed when the url from chromium is sent to port tcp:1200. So easier method to test this part is to send some nicely formatted data to the tcp port from anywhere.
With some malformed content the scp may fail but i think the execution continues to the two ssh remote executors at the end.
I need to try this just out of curiosity,,

Reply:
Sure, it would be great if you could find time to test it. There are two openpdf scripts. One is in the guivm and the other one is in the chromium-vm which just sends a string over tcp using netcat. It's used by the xdg handler but it can be executed manually. The script should stop because of the '-e' option which means “if any pipeline ever ends with a non-zero ('error') exit status, terminate the script immediately” but I'm not the biggest bash expert to be honest.